packages/core/src/routes/interaction/verifications/mfa-payload-verification.ts

```diff
@@ -128,6 +130,26 @@ const verifyBindBackupCode = async (
   return { type, codes };
 };
 
+const verifyBackupCode = async (
+  mfaVerifications: User['mfaVerifications'],
+  payload: BackupCodeVerificationPayload
+): Promise<VerifyMfaResult> => {
+  const backupCode = mfaVerifications.find(
+    (mfa): mfa is MfaVerificationBackupCode => mfa.type === MfaFactor.BackupCode
+  );
+
+  assertThat(backupCode, 'session.mfa.invalid_backup_code');
```

Comment:
nit: not sure whether the error message is accurate, this seems an error like "back up code record not found"?

Reply:
We won't leak the details, not found IS invalid.